selfdrive/debug/print_docs_diff.py

```diff
@@ -1,117 +1,123 @@
-    matches = difflib.get_close_matches(new.name, [b.name for b in base_cars], cutoff=0.)
-    if not len(matches) or matches[0] in [c[1].name for c in changes]:
-      additions.append(new)
-    else:
-      changes.append((new, next(car for car in base_cars if car.name == matches[0])))
-
-  # Removal if base car not in changes
-  removals = [b for b in base_cars if b.name not in [c[1].name for c in changes]]
-  return changes, additions, removals
-
-
-def build_column_diff(base_car, new_car):
-  row_builder = []
-  for column in Column:
-    base_column = base_car.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG)
-    new_column = new_car.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG)
-
-    if base_column != new_column:
-      row_builder.append(f"{base_column} {ARROW_SYMBOL} {new_column}")
-    else:
-      row_builder.append(new_column)
-
-  return format_row(row_builder)
-
-
-def format_row(builder):
-  return "|" + "|".join(builder) + "|"
-
-
-def print_car_docs_diff(path):
-  base_car_docs = defaultdict(list)
-  new_car_docs = defaultdict(list)
-
-  for car in load_base_car_docs(path):
-    base_car_docs[car.car_fingerprint].append(car)
-  for car in get_all_car_docs():
-    new_car_docs[car.car_fingerprint].append(car)
-
-  # Add new platforms to base cars so we can detect additions and removals in one pass
-  base_car_docs.update({car: [] for car in new_car_docs if car not in base_car_docs})
-
-  changes = defaultdict(list)
-  for base_car_model, base_cars in base_car_docs.items():
-    # Match car info changes, and get additions and removals
-    new_cars = new_car_docs[base_car_model]
-    car_changes, car_additions, car_removals = match_cars(base_cars, new_cars)
-
-    # Removals
-    for car_docs in car_removals:
-      changes["removals"].append(format_row([car_docs.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG) for column in Column]))
-
-    # Additions
-    for car_docs in car_additions:
-      changes["additions"].append(format_row([car_docs.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG) for column in Column]))
-
-    for new_car, base_car in car_changes:
-      # Column changes
-      row_diff = build_column_diff(base_car, new_car)
-      if ARROW_SYMBOL in row_diff:
-        changes["column"].append(row_diff)
-
-      # Detail sentence changes
-      if base_car.detail_sentence != new_car.detail_sentence:
-        changes["detail"].append(f"- Sentence for {base_car.name} changed!\n" +
+def download_file(url, filename):
+  response = requests.get(url)
+  with open(filename, 'wb') as file:
+    file.write(response.content)
+
+def delete_file(filename):
+  if os.path.exists(filename):
+    os.remove(filename)
+
+def column_change_format(line1, line2):
+  info1, info2 = line1.split('|'), line2.split('|')
+  return "|".join([f"{i1} {ARROW_SYMBOL} {i2}|" if i1 != i2 else f"{i1}|" for i1, i2 in zip(info1, info2, strict=True)])
+
+def get_detail_sentence(data):
```

Comment:
Maybe storing the sentence as a hoverable tooltip/alt image text would be a good idea?

I'm fine with a tiny bit of parsing logic over re-creating the sentence. Remember, any changes to the website logic would then need to be mirrored here, which is very error prone